Basic-Car-Maintenance/Shared/Odometer/ViewModels/OdometerViewModel.swift

```diff
@@ -0,0 +1,125 @@
+                    .addDocument(from: readingToAdd)
+                
+                var reading = odometerReading
+                let documentId = documentReference.documentID
+                reading.id = documentId
+                
+                readings.append(reading)
+                
+                errorMessage = ""
+                isShowingAddOdometerReading = false
+            } catch {
+                showAddErrorAlert.toggle()
+                errorMessage = error.localizedDescription
+            }
+        }
+    }
+    
+    func getOdometerReadings() async {
+        if let uid = authenticationViewModel.user?.uid {
+            let db = Firestore.firestore()
+            let docRef = db.collection("odometer_readings").whereField("userID", isEqualTo: uid)
+            
+            let querySnapshot = try? await docRef.getDocuments()
+            
+            var readings = [OdometerReading]()
+            
+            if let querySnapshot {
+                for document in querySnapshot.documents {
+                    if let reading = try? document.data(as: OdometerReading.self) {
+                        readings.append(reading)
+                    }
+                }
+                self.readings = readings
+            }
+        }
+    }
+    
+    func getVehicles() async {
+        if let uid = authenticationViewModel.user?.uid {
+            let db = Firestore.firestore()
+            let docRef = db.collection("vehicles").whereField("userID", isEqualTo: uid)
+            
+            let querySnapshot = try? await docRef.getDocuments()
+            
+            var vehicles = [Vehicle]()
+            
+            if let querySnapshot {
+                for document in querySnapshot.documents {
+                    if let vehicle = try? document.data(as: Vehicle.self) {
+                        vehicles.append(vehicle)
+                    }
+                }
+                
+                self.vehicles = vehicles
+            }
+        }
+    }
+}
+
+extension OdometerViewModel {
+    enum SortOption: Int, CaseIterable, Identifiable {
+        case oldestToNewest = 0
+        case newestToOldest = 1
+        case custom = 2
+        
+        var id: Int {
+            rawValue
+        }
+        
+        var label: LocalizedStringResource {
+            switch self {
+            case .oldestToNewest:
+                LocalizedStringResource(
+                    "Oldest to Newest",
+                    comment: "Sorting option that displays older items first.")
+            case .newestToOldest:
+                LocalizedStringResource(
+                    "Newest to Oldest",
+                    comment: "Sorting option that displays newer items first.")
+            case .custom:
+                LocalizedStringResource(
+                    "Custom",
+                    comment: "Sorting option that sorts items according to the user's preferences.")
+            }
+        }
+    }
+}
```

Comment:
We can remove all of this for now

Reply:
fixed in https://github.com/mikaelacaron/Basic-Car-Maintenance/pull/175/commits/6ee9780bdab3e2f73b510c75830667c6e130e016